bqemulatormanager/schema.py

```diff
@@ -1,48 +1,69 @@
 import os
-from typing import List
+from typing import List, Optional, Protocol
 
 import yaml
 from google.cloud import bigquery
+from google.cloud.bigquery.table import Table
+
+
+class SchemaError(Exception):
+    pass
+
+
+class BigQueryClient(Protocol):
+    def get_table(self, _table: str) -> Table:  # noqa: U101
+        pass
 
 
 class SchemaManager:
-    def __init__(self, master_path: str = "master_schema.yaml", client=None):
-        self.client = client
-        self.master_path = master_path
-        self.change_flg = False
-        if os.path.isfile(master_path):
-            with open(master_path) as f:
+    """
+    SchemaManager manage schema of tables in emulator.
+
+    params:
+        schema_file_path:
+            path to yaml file which contains schema of tables in emulator.
+
+        client:
+            client of production BigQuery. If you set this, SchemaManager will fetch schema from production BigQuery.
+    """
+
+    def __init__(self, schema_file_path: Optional[str] = "master_schema.yaml", client: Optional[BigQueryClient] = None):
```

Comment:
[IMO, nits] Current default schema file name is too common to hit the other kind of file unintentionally.  How about change it to `bqem_master_schema.yaml`?